libs/core/langchain_core/tools.py

```diff
@@ -1482,9 +1481,8 @@ def _prep_run_args(
 ) -> Tuple[Union[str, Dict], Dict]:
     config = ensure_config(config)
     if _is_tool_call(input):
-        input_copy = copy.deepcopy(input)
-        tool_call_id: Optional[str] = cast(ToolCall, input_copy)["id"]
-        tool_input: Union[str, dict] = cast(ToolCall, input_copy)["args"]
+        tool_call_id: Optional[str] = cast(ToolCall, input)["id"]
+        tool_input: Union[str, dict] = cast(ToolCall, input)["args"].copy()
```

Comment:
should we try/except deepcopy instead?

Reply:
maybe i take it back, are we sure we need to copy at all?

Reply:
If I remove `.copy` then this test fails because the config gets added to the tool call args: https://github.com/langchain-ai/langchain/blob/df786087412e40d3c473d61af4e133373e2eaabe/libs/core/tests/unit_tests/test_tools.py#L981-L988

This will generally mutate the AIMessage holding the tool calls as well.

lmk if you see a better solution and can revisit.

Reply:
should we update how we inject args instead of copyign here?